src/Microsoft.Azure.WebJobs.Host/Scale/ScaleManager.cs

```diff
@@ -0,0 +1,297 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Manages scale monitoring operations.
+    /// </summary>
+    public class ScaleManager
```

Comment:
Does this type need to be public? 

Should we define (and register) an interface for this?

Reply:
We had defined an interface, but I just made it a public service type, similar to the decision I made for ConcurrencyManager.